From bdd13f6be4b588568683a1ab54f421fc6a636dbb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Sat, 22 Feb 2014 19:26:27 -0500
Subject: [PATCH 4/4] Remove dead lines in various places

As pointed-out by clang -Wunreachable-code.

No behaviour changes.
---
 src/bootchart/bootchart.c             |  1 -
 src/bus-proxyd/bus-proxyd.c           |  2 --
 src/core/service.c                    |  4 +---
 src/journal/journal-file.c            |  2 --
 src/journal/journald-server.c         | 18 +++++++-----------
 src/journal/sd-journal.c              |  2 --
 src/libsystemd/sd-event/sd-event.c    |  2 --
 src/libsystemd/sd-rtnl/rtnl-message.c | 12 +++++++-----
 src/login/logind.c                    |  2 --
 src/shared/install.c                  |  2 --
 src/shared/time-dst.c                 |  4 ++--
 src/udev/scsi_id/scsi_id.c            |  1 -
 src/udev/udevadm-info.c               |  3 +--
 13 files changed, 18 insertions(+), 37 deletions(-)

diff --git a/src/bootchart/bootchart.c b/src/bootchart/bootchart.c
index 305a31a..9f72d98 100644
--- a/src/bootchart/bootchart.c
+++ b/src/bootchart/bootchart.c
@@ -220,7 +220,6 @@ static int parse_args(int argc, char *argv[]) {
                         fprintf(stderr, " --help,      -h          Display this message\n");
                         fprintf(stderr, "See bootchart.conf for more information.\n");
                         exit (EXIT_SUCCESS);
-                        break;
                 default:
                         break;
                 }
diff --git a/src/bus-proxyd/bus-proxyd.c b/src/bus-proxyd/bus-proxyd.c
index b9e4d9d..80f83e7 100644
--- a/src/bus-proxyd/bus-proxyd.c
+++ b/src/bus-proxyd/bus-proxyd.c
@@ -774,8 +774,6 @@ int main(int argc, char *argv[]) {
                 }
         }
 
-        r = 0;
-
 finish:
         sd_bus_flush(a);
         sd_bus_flush(b);
diff --git a/src/core/service.c b/src/core/service.c
index 6977774..6de24ec 100644
--- a/src/core/service.c
+++ b/src/core/service.c
@@ -1436,9 +1436,7 @@ static int service_search_main_pid(Service *s) {
                 log_warning_unit(UNIT(s)->id,
                                  "Failed to watch PID "PID_FMT" from service %s",
                                  pid, UNIT(s)->id);
-                return r;
-
-        return 0;
+        return r;
 }
 
 static void service_set_state(Service *s, ServiceState state) {
diff --git a/src/journal/journal-file.c b/src/journal/journal-file.c
index 23c4d28..5876733 100644
--- a/src/journal/journal-file.c
+++ b/src/journal/journal-file.c
@@ -2210,8 +2210,6 @@ int journal_file_move_to_entry_by_monotonic_for_data(
 
                 z = q;
         }
-
-        return 0;
 }
 
 int journal_file_move_to_entry_by_seqnum_for_data(
diff --git a/src/journal/journald-server.c b/src/journal/journald-server.c
index 96ef66d..2a90b44 100644
--- a/src/journal/journald-server.c
+++ b/src/journal/journald-server.c
@@ -1131,15 +1131,13 @@ int process_datagram(sd_event_source *es, int fd, uint32_t revents, void *userda
                 union {
                         struct cmsghdr cmsghdr;
 
-                        /* We use NAME_MAX space for the
-                         * SELinux label here. The kernel
-                         * currently enforces no limit, but
-                         * according to suggestions from the
-                         * SELinux people this will change and
-                         * it will probably be identical to
-                         * NAME_MAX. For now we use that, but
-                         * this should be updated one day when
-                         * the final limit is known.*/
+                        /* We use NAME_MAX space for the SELinux label
+                         * here. The kernel currently enforces no
+                         * limit, but according to suggestions from
+                         * the SELinux people this will change and it
+                         * will probably be identical to NAME_MAX. For
+                         * now we use that, but this should be updated
+                         * one day when the final limit is known.*/
                         uint8_t buf[CMSG_SPACE(sizeof(struct ucred)) +
                                     CMSG_SPACE(sizeof(struct timeval)) +
                                     CMSG_SPACE(sizeof(int)) + /* fd */
@@ -1216,8 +1214,6 @@ int process_datagram(sd_event_source *es, int fd, uint32_t revents, void *userda
 
                 close_many(fds, n_fds);
         }
-
-        return 0;
 }
 
 static int dispatch_sigusr1(sd_event_source *es, const struct signalfd_siginfo *si, void *userdata) {
diff --git a/src/journal/sd-journal.c b/src/journal/sd-journal.c
index 3740a9a..ef455e9 100644
--- a/src/journal/sd-journal.c
+++ b/src/journal/sd-journal.c
@@ -2293,8 +2293,6 @@ _public_ int sd_journal_process(sd_journal *j) {
                         l -= step;
                 }
         }
-
-        return determine_change(j);
 }
 
 _public_ int sd_journal_wait(sd_journal *j, uint64_t timeout_usec) {
diff --git a/src/libsystemd/sd-event/sd-event.c b/src/libsystemd/sd-event/sd-event.c
index 253923d..db76433 100644
--- a/src/libsystemd/sd-event/sd-event.c
+++ b/src/libsystemd/sd-event/sd-event.c
@@ -1814,8 +1814,6 @@ static int process_signal(sd_event *e, uint32_t events) {
                 if (r < 0)
                         return r;
         }
-
-        return 0;
 }
 
 static int source_dispatch(sd_event_source *s) {
diff --git a/src/libsystemd/sd-rtnl/rtnl-message.c b/src/libsystemd/sd-rtnl/rtnl-message.c
index 8c20b8e..dc74947 100644
--- a/src/libsystemd/sd-rtnl/rtnl-message.c
+++ b/src/libsystemd/sd-rtnl/rtnl-message.c
@@ -480,7 +480,7 @@ int sd_rtnl_message_append_u16(sd_rtnl_message *m, unsigned short type, uint16_t
                                 break;
                         else
                                 return -ENOTSUP;
-                        break;
+
                 default:
                         return -ENOTSUP;
         }
@@ -932,10 +932,6 @@ int socket_read_message(sd_rtnl *nl, sd_rtnl_message **ret) {
 
         if (k > 0)
                 switch (m->hdr->nlmsg_type) {
-                        struct ifinfomsg *ifi;
-                        struct ifaddrmsg *ifa;
-                        struct rtmsg *rtm;
-
                         /* check that the size matches the message type */
                         case NLMSG_ERROR:
                                 if (m->hdr->nlmsg_len < NLMSG_LENGTH(sizeof(struct nlmsgerr)))
@@ -948,6 +944,8 @@ int socket_read_message(sd_rtnl *nl, sd_rtnl_message **ret) {
                                 if (m->hdr->nlmsg_len < NLMSG_LENGTH(sizeof(struct ifinfomsg)))
                                         k = -EIO;
                                 else {
+                                        struct ifinfomsg *ifi;
+
                                         ifi = NLMSG_DATA(m->hdr);
                                         UPDATE_RTA(m, IFLA_RTA(ifi));
                                 }
@@ -958,6 +956,8 @@ int socket_read_message(sd_rtnl *nl, sd_rtnl_message **ret) {
                                 if (m->hdr->nlmsg_len < NLMSG_LENGTH(sizeof(struct ifaddrmsg)))
                                         k = -EIO;
                                 else {
+                                        struct ifaddrmsg *ifa;
+
                                         ifa = NLMSG_DATA(m->hdr);
                                         UPDATE_RTA(m, IFA_RTA(ifa));
                                 }
@@ -968,6 +968,8 @@ int socket_read_message(sd_rtnl *nl, sd_rtnl_message **ret) {
                                 if (m->hdr->nlmsg_len < NLMSG_LENGTH(sizeof(struct rtmsg)))
                                         k = -EIO;
                                 else {
+                                        struct rtmsg *rtm;
+
                                         rtm = NLMSG_DATA(m->hdr);
                                         UPDATE_RTA(m, RTM_RTA(rtm));
                                 }
diff --git a/src/login/logind.c b/src/login/logind.c
index 3a514bb..10f61ab 100644
--- a/src/login/logind.c
+++ b/src/login/logind.c
@@ -1078,8 +1078,6 @@ int manager_run(Manager *m) {
                 if (r < 0)
                         return r;
         }
-
-        return 0;
 }
 
 static int manager_parse_config_file(Manager *m) {
diff --git a/src/shared/install.c b/src/shared/install.c
index 1f9d7e2..f57b94d 100644
--- a/src/shared/install.c
+++ b/src/shared/install.c
@@ -469,8 +469,6 @@ static int find_symlinks_fd(
                                 return 1;
                 }
         }
-
-        return r;
 }
 
 static int find_symlinks(
diff --git a/src/shared/time-dst.c b/src/shared/time-dst.c
index 65e2998..ceca2fa 100644
--- a/src/shared/time-dst.c
+++ b/src/shared/time-dst.c
@@ -207,8 +207,8 @@ read_again:
                 if (type_idxs[i] >= num_types)
                         return -EINVAL;
 
-        if ((BYTE_ORDER != BIG_ENDIAN && (sizeof(time_t) == 4 || trans_width == 4)) ||
-            (BYTE_ORDER == BIG_ENDIAN && sizeof(time_t) == 8 && trans_width == 4)) {
+        if (BYTE_ORDER == BIG_ENDIAN ? sizeof(time_t) == 8 && trans_width == 4
+                                     : sizeof(time_t) == 4 || trans_width == 4) {
                 /* Decode the transition times, stored as 4-byte integers in
                    network (big-endian) byte order.  We work from the end of
                    the array so as not to clobber the next element to be
diff --git a/src/udev/scsi_id/scsi_id.c b/src/udev/scsi_id/scsi_id.c
index 03bd3a9..5511df8 100644
--- a/src/udev/scsi_id/scsi_id.c
+++ b/src/udev/scsi_id/scsi_id.c
@@ -396,7 +396,6 @@ static int set_options(struct udev *udev,
                 case 'V':
                         printf("%s\n", VERSION);
                         exit(0);
-                        break;
 
                 case 'x':
                         export = true;
diff --git a/src/udev/udevadm-info.c b/src/udev/udevadm-info.c
index 64bb537..8145e6f 100644
--- a/src/udev/udevadm-info.c
+++ b/src/udev/udevadm-info.c
@@ -498,8 +498,7 @@ static int uinfo(struct udev *udev, int argc, char *argv[])
                         print_record(device);
                         break;
                 default:
-                        fprintf(stderr, "unknown query type\n");
-                        break;
+                        assert_not_reached("unknown query type");
                 }
                 break;
         case ACTION_ATTRIBUTE_WALK:
-- 
1.9.0

